### ISSUE TITLE
e.reduce is not a function

### ISSUE THREAD

**Issue by maxibanki** [Microsoft/vscode#39233]
- VSCode Version: Code - Insiders 1.19.0-insider (3c470706ded48fc75c05f86bc4f9c7b0f4d6c708, 2017-11-27T05:14:17.223Z)
- OS Version: Windows_NT x64 6.1.7601
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
markdown-checkbox|bie|0.0.5
jshint|dba|0.10.15
vscode-eslint|dba|1.4.3
githistory|don|0.2.3
xml|Dot|1.9.2
gitlens|eam|6.2.0
tslint|eg2|1.0.24
vscode-abap|lar|0.1.0
Go|luk|0.6.67
postscript|max|1.0.0
code-beautifier|mic|1.7.2
debugger-for-chrome|msj|3.5.0
vscode-docker|Pet|0.0.22
vscode-icons|rob|7.18.1
vscode-table-formatter|shu|1.2.1
addDocComments|ste|0.0.8
html-preview-vscode|tht|0.1.1
vscode-import-cost|wix|2.5.1

(1 theme extensions excluded)

---

Steps to Reproduce:

1. Open extension sidebar
2. Click on:
![image](https://user-images.githubusercontent.com/17984549/33307365-80289fd4-d416-11e7-9613-331949945216.png)

Then this appears:

![image](https://user-images.githubusercontent.com/17984549/33307484-f147189e-d416-11e7-8706-23581ce8aa13.png)

DevTools:

```

e.reduce is not a function: TypeError: e.reduce is not a function,    at e.joinErrors (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:216:359),    at file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:210:792,    at Object.g [as _notify] (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:63:21),    at Object.enter (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:66:264),    at n.Class.derive._creator._run (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:67:573),    at n.Class.derive._creator._chainedError (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:66:944),    at n (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:64:794),    at n.Class.define.cancel.then (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:68:360),    at Object.enter (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:64:863),    at n.Class.derive._creator._run (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:67:573),    at n.Class.derive._creator._setCompleteValue (file:///C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:67:267)
e.doShow @ messageService.ts:126
e.show @ messageService.ts:105
(anonymous) @ extensionsList.ts:91
e.fire @ event.ts:142
(anonymous) @ actionbar.ts:399
e.fire @ event.ts:142
(anonymous) @ actions.ts:221
v @ winjs.base.raw.js:1209
enter @ winjs.base.raw.js:901
_run @ winjs.base.raw.js:1068
_chainedError @ winjs.base.raw.js:1031
n @ winjs.base.raw.js:736
v @ winjs.base.raw.js:1209
enter @ winjs.base.raw.js:901
_run @ winjs.base.raw.js:1068
_error @ winjs.base.raw.js:1041
e.doRequest.n.TPromise.t.handlers.(anonymous function) @ ipc.ts:254
e.onMessage @ ipc.ts:319
(anonymous) @ ipc.ts:206
e.fire @ event.ts:142
(anonymous) @ ipc.net.ts:81
emitOne @ events.js:96
emit @ events.js:191
readableAddChunk @ _stream_readable.js:178
Readable.push @ _stream_readable.js:136
onread @ net.js:560
```

Shared Process logs:

```
/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:78 WARNING: Promise with no error callback:undefined
/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:78 Error: Promise did not implement oncancel
    at n.Class.derive._oncancel._cancelAction (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:69)
    at Object.enter (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:65)
    at n.Class.derive._oncancel._run (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:67)
    at n.Class.derive._oncancel.cancel (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:66)
    at Object.enter (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:65)
    at n.Class.derive._creator._run (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:67)
    at n.Class.derive._creator.cancel (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:66)
    at n.Class.derive._creator._cancelAction (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:67)
    at Object.enter (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:65)
    at n.Class.derive._creator._run (/C:/Programme/VSCodeInsider/resources/app/out/vs/code/electron-browser/sharedProcessMain.js:67)
```

Reproduces without extensions: No. Without extensions the update was successful. => can't reproduce it anymore. But I've started VS Code again before I had ran it without extensions. There the error occured again.

**Comment by kieferrm** [comment]
@mjbvz didn't you look into similar issues before?

**Comment by mjbvz** [comment]
@sandy081 Looks like #35539. Can you please take a look?

**Comment by sandy081** [comment]
This was caused by e2afd7d429f83e86b42fe905c599222da09c0f7f and was fixed few days back - 9f5614452583e3bd4c62b67acf927d97b6544f0e 

Please try the latest insiders. Thanks.
